src/ui/components/anchor/AnchorContext.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { createContext, useContext, useRef, RefObject, useMemo, useCallback } from 'react'
+import { ScrollView, View } from 'react-native'
+
+import { AnchorName } from 'ui/components/anchor/anchor-name'
+
+type AnchorContextType = {
+  registerAnchor: (name: AnchorName, ref: RefObject<View>) => void
+  scrollToAnchor: (name: AnchorName) => void
+}
+
+const AnchorContext = createContext<AnchorContextType | undefined>(undefined)
+
+type AnchorProviderProps = {
+  scrollViewRef: RefObject<ScrollView>
+  handleCheckScrollY(): number
+  children: React.ReactNode
+}
+
+export const AnchorProvider = ({
+  scrollViewRef,
+  handleCheckScrollY,
+  children,
+}: AnchorProviderProps) => {
+  const anchors = useRef<Record<AnchorName, RefObject<View>>>(
+    {} as Record<AnchorName, RefObject<View>>
+  )
```

Comment:
si on peu éviter les `as`, je trouve ça mieux

```suggestion
  const trouveMoiUnNom: Record<AnchorName, RefObject<View>> = {}
  const anchors = useRef<typeof trouveMoiUnNom>(
    trouveMoiUnNom
  )
```